README.md

```diff
@@ -1,18 +1,44 @@
 # Supabase Edge Runtime
 
-A web server based on [Deno](https://deno.land) runtime, capable of running JavaScript, TypeScript, and WASM services.
+A Javascript runtime based on [Deno](https://deno.land), capable of running JavaScript, TypeScript, and WASM services.
 
 You can use it to:
 
-* Locally test and self-host Supabase's Edge Functions (or any Deno Edge Function)
+* Locally test and self-host Supabase's Edge Functions (or any Deno Function)
 * As a programmable HTTP Proxy: You can intercept / route HTTP requests
 
 **WARNING: Beta Software. There will be breaking changes to APIs / Configuration Options**
 
+## Architecture
+
+The edge runtime can be divided into two runtimes with different purposes.
+- Main runtime:
+  - An instance for the _main runtime_ is responsible for proxying the transactions to the _user runtime_.
+  - The main runtime is meant to be an entry point before running user functions, where you can authentication, etc. before calling the user function.
+  - Has no user-facing limits
+- User runtime:
+  - An instance for the _user runtime_ is responsible for executing users' code.
+  - Limits are required to be set such as: Memory and Timeouts
+  - Environmental Variables are readonly.
```

Comment:
Envirtonmental Variables are readonly in both cases. The differences are:
* Main Worker has access to OS environment variables
* User Worker only get access to environment variables provided by the Main Worker.